Makefile

```diff
@@ -1,157 +1,282 @@
-# A Self-Documenting Makefile: http://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
 
-export PATH := $(abspath bin/):${PATH}
+# Image URL to use all building/pushing image targets
+IMG ?= ghcr.io/bank-vaults/vault-operator:dev
 
-CONTAINER_IMAGE_REF = ghcr.io/bank-vaults/vault-operator:dev
-
-# Dependency versions
+# Tool chain
 GOLANGCI_VERSION = 1.53.3
 LICENSEI_VERSION = 0.8.0
 KIND_VERSION = 0.20.0
 KURUN_VERSION = 0.7.0
 CODE_GENERATOR_VERSION = 0.27.1
-CONTROLLER_GEN_VERSION = 0.12.0
 HELM_DOCS_VERSION = 1.11.0
+KUSTOMIZE_VERSION = 5.0.1
+CONTROLLER_TOOLS_VERSION = 0.12.0
+ENVTEST_K8S_VERSION = 1.27.1
 
-.PHONY: up
-up: ## Start development environment
-	kind create cluster
-
-.PHONY: stop
-stop: ## Stop development environment
-	# TODO: consider using k3d instead
-	kind delete cluster
+# Acceptance test default data
+TEST_VAULT_VERSION = 1.13.3
+TEST_BANK_VAULTS_VERSION = 1.19.0
 
-.PHONY: down
-down: ## Destroy development environment
-	kind delete cluster
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
 
-.PHONY: build
-build: ## Build binary
-	@mkdir -p build
-	go build -race -o build/manager ./cmd/manager
+# Setting SHELL to bash allows bash commands to be executed by recipes.
+# Options are set to exit when a recipe line exits non-zero or a piped command fails.
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
 
-.PHONY: run
-run: ## Run the operator locally talking to a Kubernetes cluster
-	kubectl replace -f deploy/default/crd.yaml || kubectl create -f deploy/default/crd.yaml
-	kubectl apply -f deploy/default/rbac.yaml
-	OPERATOR_NAME=vault-dev go run cmd/manager/main.go -verbose
+##@ General
 
-.PHONY: clean
-clean: ## Clean operator resources from a Kubernetes cluster
-	kubectl delete -f deploy/default/crd.yaml
-	kubectl delete -f deploy/default/rbac.yaml
+.PHONY: all
+all: build
 
-.PHONY: artifacts
-artifacts: container-image helm-chart
-artifacts: ## Build artifacts
-
-.PHONY: container-image
-container-image: ## Build container image
-	docker build -t ${CONTAINER_IMAGE_REF} .
-
-.PHONY: helm-chart
-helm-chart: ## Build Helm chart
-	@mkdir -p build
-	helm package -d build/ deploy/charts/vault-operator
+.PHONY: help
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
 
-.PHONY: check
-check: test lint ## Run checks (tests and linters)
+##@ Development
 
-.PHONY: test
-test: ## Run tests
-	go test -race -v ./...
+.PHONY: fmt
+fmt: golangci-lint ## Run go fmt against code
```

Comment:
golangci-lint is an unnecessary dependency when using Nix. Also, do not call go fmt out here as we could add any kind of formatter.

Reply:
Let's take a look at that and see if we can resolve that problem. I want to make Nix a first-class citizen, but mostly optional (ie. allow running make deps to download everything). I'm okay with not using Nix being less comfortable for people.

Reply:
That is okay. Another thing to consider is the pipeline, nix setup takes considerable amount of time, perhaps if it is possible to add caching with nix. I have added a ticket for this #105